vaadin-date-picker-flow-parent/vaadin-date-picker-flow-integration-tests/src/test/java/com/vaadin/flow/component/datepicker/DatePickerFormatIT.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Test
+    public void testWithMultipleFormats() {
+        Assert.assertEquals("2018.05.13", getInputValue(MULTIPLE_FORMAT));
+    }
+
+    @Test
+    public void testChangeBetweenFormatsFormatsInNewFormat() {
+        String id = CHANGE_BETWEEN_FORMATS;
+
+        Assert.assertEquals("13.2018.05", getInputValue(id));
+        $("button").id(CHANGE_TO_FORMAT_BTN).click();
+        Assert.assertEquals("5/13/18", getInputValue(id));
+    }
+
+    @Test
+    public void testChangeBetweenFormatsParsesInNewFormat() {
+        $("button").id(CHANGE_TO_FORMAT_BTN).click();
+
+        submitValue(CHANGE_BETWEEN_FORMATS, "2/27/21");
+
+        TestBenchElement output = $("span").id(CHANGE_BETWEEN_FORMATS_OUTPUT);
+
+        Assert.assertEquals("2021-02-27", output.getText());
+    }
+
+    @Test
+    public void testFormatsWithSetLocalShouldFormatWithCustomFormat() {
+        Assert.assertEquals(getInputValue(FORMAT_AND_SET_LOCALE), "2018/05/13");
+    }
+
+    @Test
+    public void testFormatsWithSetLocalShouldParseWithCustomFormat() {
+        submitValue(FORMAT_AND_SET_LOCALE, "1999/07/15");
+
+        TestBenchElement output = $("span").id(FORMAT_AND_SET_LOCALE_OUTPUT);
+
+        Assert.assertEquals("1999-07-15", output.getText());
+    }
+
+    @Test
+    public void testSetFormatAfterSetLocale() {
+        Assert.assertEquals(getInputValue(CHANGE_FROM_SET_LOCALE),
+                "2018/05/13");
+    }
+
+    @Test
+    public void testNullFormatShouldFormatWithLocaleFormat() {
+        String id = NULL_FORMAT;
+
+        Assert.assertEquals("13 2018 05", getInputValue(id));
+        $("button").id(SET_FORMAT_TO_NULL_BUTTON).click();
+        Assert.assertEquals("13.5.2018", getInputValue(id));
+    }
+
+    @Test
+    public void testNullFormatShouldParseWithLocaleFormat() {
+        $("button").id(SET_FORMAT_TO_NULL_BUTTON).click();
+
+        submitValue(NULL_FORMAT, "15.07.1999");
+
+        TestBenchElement output = $("span").id(NULL_FORMAT_OUTPUT);
+
+        Assert.assertEquals("1999-07-15", output.getText());
+    }
+
+    /**
+     * Verify datePickerConnector.setLocale is not called with null parameter
+     */
+    @Test
+    public void testNullFormatShouldNotLogBrowserError() {
+        $("button").id(SET_FORMAT_TO_NULL_BUTTON).click();
+
+        Assert.assertFalse(
+                getLogEntries(Level.SEVERE).stream().findAny().isPresent());
+    }
+
+    @Test
+    public void testParsingSingleFormat() {
+        submitValue(PARSING_SINGLE_FORMAT, "2020/10/23");
+
+        TestBenchElement output = $("span").id(PARSING_SINGLE_FORMAT_OUTPUT);
+        Assert.assertEquals("2020-10-23", output.getText());
+    }
+
+    @Test
+    public void testWithClientSideChange() {
+        submitValue(CLIENT_CHANGE, "23-10-2020");
+
+        TestBenchElement output = $("span").id(CLIENT_CHANGE_OUTPUT);
+        Assert.assertEquals("2020-10-23", output.getText());
```

Comment:
This tests the fallback feature, and specifically that the last format parses the input. We can add another input/assertion to this test to check that the second format works as well.

Reply:
done